wrapanapi/systems/redfish.py

```diff
@@ -13,18 +15,54 @@ class RedfishSystem(System):
         hostname: The hostname of the system.
         username: The username to connect with.
         password: The password to connect with.
+        security_protocol: The security protocol to be used for connecting with
+            the API. Expected values: 'Non-SSL', 'SSL', 'SSL without validation'
     """
+    # Power states as defined in Redfish
+    RF_POWER_STATE_ON = 'On'
+    RF_POWER_STATE_OFF = 'Off'
+    RF_POWER_STATE_POWERING_ON = 'PoweringOn'
+    RF_POWER_STATE_POWERING_OFF = 'PoweringOff'
+    # Power states in CloudForms
```

Comment:
These should be defined in the CloudForms classes for modeling RedFish, not here in wrapanapi.

If anything, the states that RedFish returns should be mapped to VmState enumerations, as demonstrated here in the EC2System class:
https://github.com/ManageIQ/wrapanapi/blob/master/wrapanapi/systems/ec2.py#L47

Reply:
> These should be defined in the CloudForms classes for modeling RedFish, not here in wrapanapi.

Just to be clear, you are OK with the `RF_POWER_STATE_*` definitions (I did them because I've seen [your comment](https://github.com/ManageIQ/wrapanapi/pull/294#discussion_r207254977) in what I thought was a similar situation), but I should move the mapping CF ones somewhere else?

> If anything, the states that RedFish returns should be mapped to VmState enumerations, as demonstrated here in the EC2System class

These states are not about VMs, they are about physical servers. I don't think mixing the two would be a good idea. But unlike the VM representations, there is no such common representation that would be defined or even prescribed for the physical servers. So I wrote the mapping assuming something that isn't there - the providers actually push the lower-case version of whatever state is there and that appears on the GUI.

Assuming that we decide to establish a common set of states, would it then make sense for us to create a `wrapanap/entities/physical_server.py` with a `PhysicalServerState` enumeration?

Reply:
You're probably right that its a bad idea to conflate VM States and physical server states, and are right about creating a base entity class for servers. I would recommend just using `server.py` and ServerState class.

Then a RedfishServer class can inherit this entity class. 

Note how `state_map` is used in `entities.vm.Vm`, and the `_api_state_to_vmstate()` method resolves the map to a constant state.  The system-specific VM classes then implement `_get_state`, calling `_api_state_to_vmstate`  (note this is an abstractmethod for `Vm`).

The pattern described above should be mirrored in a new entity class, which may change how you've defined `FR_POWER_STATE_*` here.

So by extending the entity base class into a new server entity class, you can establish the same structure and pattern that is used by the VM's to manage server states in a consistent manner across different provider types that host servers.

I suggest putting this in a 'Server' class and not 'physical_server' because this same class could be applied to hosts for other providers (rhevm, openstack, virtualcenter, etc)

Reply:
If the states are essentially the same, I think we'd be better off abstracting VmState a bit more to a generic 'PowerState' class, and then extending that as a 'mixin' type class where we need to add some state unique to a Vm, or a server, or a template, or a switch, etc.